src/adapters/aol.js

```diff
@@ -1,183 +1,141 @@
+    };
   }
 
-  /**
-   * @private Add a succesful bid response for aol
-   * @param {ADTECHResponse} response the response for the bid
-   * @param {ADTECHContext} context the context passed from aol
-   */
-  function _addBid(response, context) {
-    var bid = bidsMap[context.alias];
-    var cpm;
+  function _buildPubapiUrl(bid) {
+    const params = bid.params;
+    const serverParam = params.server;
+    var regionParam = params.region || 'us';
+    var server;
 
-    if (!bid) {
-      utils.logError('mismatched bid: ' + context.placement, ADTECH_BIDDER_NAME, context);
-      return;
+    if (!SERVER_MAP.hasOwnProperty(regionParam)) {
+      console.warn(`Unknown region '${regionParam}' for AOL bidder.`);
+      regionParam = 'us'; // Default region.
     }
 
-    cpm = response.getCPM();
-    if (cpm === null || isNaN(cpm)) {
-      return _addErrorBid(response, context);
+    if (serverParam) {
+      server = serverParam;
+    } else {
+      server = SERVER_MAP[regionParam];
     }
 
-    // clean up--we no longer need to store the bid
-    delete bidsMap[context.alias];
-
-    var bidResponse = bidfactory.createBid(1);
-    var ad = response.getCreative();
-    if (typeof response.getPixels() !== 'undefined') {
-      ad += response.getPixels();
-    }
-    bidResponse.bidderCode = ADTECH_BIDDER_NAME;
-    bidResponse.ad = ad;
-    bidResponse.cpm = cpm;
-    bidResponse.width = response.getAdWidth();
-    bidResponse.height = response.getAdHeight();
-    bidResponse.creativeId = response.getCreativeId();
+    // Set region param, used by AOL analytics.
+    params.region = regionParam;
+
+    return pubapiTemplate({
+      protocol: (document.location.protocol === 'https:') ? 'https' : 'http',
+      host: server,
+      network: params.network,
+      placement: parseInt(params.placement),
+      pageid: params.pageId || 0,
+      sizeid: params.sizeId || 0,
+      alias: params.alias || utils.getUniqueIdentifierStr(),
+      bidfloor: (typeof params.bidFloor !== 'undefined') ?
+        `;bidfloor=${params.bidFloor.toString()}` : '',
+      misc: new Date().getTime() // cache busting
+    });
+  }
 
-    // add it to the bid manager
+  function _addErrorBidResponse(bid, response = {}) {
+    var bidResponse = bidfactory.createBid(2);
+    bidResponse.bidderCode = BIDDER_CODE;
+    bidResponse.reason = response.nbr;
+    bidResponse.raw = response;
     bidmanager.addBidResponse(bid.placementCode, bidResponse);
   }
 
-  /**
-   * @private Add an error bid response for aol
-   * @param {ADTECHResponse} response the response for the bid
-   * @param {ADTECHContext} context the context passed from aol
-   */
-  function _addErrorBid(response, context) {
-    var bid = bidsMap[context.alias];
+  function _addBidResponse(bid, response) {
+    var bidData;
 
-    if (!bid) {
-      utils.logError('mismatched bid: ' + context.placement, ADTECH_BIDDER_NAME, context);
+    try {
+      bidData = response.seatbid[0].bid[0];
```

Comment:
An empty bid or "no bid" response causes this to fail which throws an error, though a response with an empty `seatbid` array appears to be common in my testing, so should be handled as other than Error.